libraries/apollo-compiler/api/apollo-compiler.api

```diff
@@ -9,17 +6,25 @@ public final class com/apollographql/apollo3/compiler/AdapterInitializer$Compani
 	public final fun serializer ()Lkotlinx/serialization/KSerializer;
 }
 
-public abstract interface class com/apollographql/apollo3/compiler/ApolloCompiler$Logger {
-	public abstract fun warning (Ljava/lang/String;)V
+public final class com/apollographql/apollo3/compiler/ApolloCompiler {
```

Comment:
Decided to make it public without an `@ApolloExperimental` marker. It's not like `apollo-compiler` has the same level of stability as `apollo-api`. We might need to break this at some point but I find this better than a bunch of noisy `@ApolloExperimental` annotations that would be basically required to use this API.